src/main/flight/imu.c

```diff
@@ -407,6 +394,73 @@ static bool imuIsAccelerometerHealthy(void)
     return (81 < accMagnitude) && (accMagnitude < 121);
 }
 
+// Calculate the dcmKpGain to use. When armed, the gain is imuRuntimeConfig.dcm_kp * 1.0 scaling.
+// When disarmed after initial boot, the scaling is set to 10.0 for the first 20 seconds to speed up initial convergence.
+// After disarming we want to quickly reestablish convergence to deal with the attitude estimation being incorrect due to a crash.
+//   - wait for a 250ms period of low gyro activity to ensure the craft is not moving
+//   - use a large dcmKpGain value for 500ms to allow the attitude estimate to quickly converge
+//   - reset the gain back to the standard setting
+float imuCalcKpGain(timeUs_t currentTimeUs, bool useAcc, float gyroX, float gyroY, float gyroZ)
+{
+    static bool lastArmState = false;
+    static timeUs_t gyroQuietPeriodTimeEnd = 0;
+    static timeUs_t attitudeResetTimeEnd = 0;
+    static bool attitudeResetCompleted = false;
+    float ret;
+    bool attitudeResetActive = false;
+
+    const bool armState = ARMING_FLAG(ARMED);
+
+    if (!armState) {
+        if (!STATE(FIXED_WING)) {  // disable for fixed wing. probably wouldn't hurt anything but not needed or tested
```

Comment:
Not sure what you mean.  I didn't see any value in this code (and not tested) in the fixed wing config.  So I structured it to have the least impact on the armed state.  I could have wrapped the whole thing with `!STATE(FIXED_WING)` but then I would have had two conditionals in the armed code path.

Reply:
It should not hurt fixed wing and it is in effect only when disarmed. There is no need for special-casing fixed wing

Reply:
Well I do have a plan for a follow-on enhancement to use this attitude reset logic when armed and transitioning from acro to level-mode to address the other major problem with the attitude estimate divergence.  So that's partially why I included the fixed-wing exclusion but I'll remove it now and address in the follow-on if appropriate.